### PR TITLE
Delete node AWS

### LINKED ISSUE #781: 2.0: AWS - Can not delete cluster
Summary of Issue:
You are unable to delete an AWS cluster

Steps to Reproduce:
1. Create an AWS cluster of any size
2. Click the delete cluster button

Expected Results:
The cluster is deleted

Actual Results:
Nothing happens

Dev Info: (fill out and add links to log files)
1. `go version`: 1.10
2. SG latest commit hash or release tag: dddedff38267d34e16763fde1955ca4dec1cc5d7
3. Number of Masters and Nodes: Any
4. cloud provider: AWS

### PR DESCRIPTION
**Pull request by stgleb**
Add delete node backend code and fix delete cluster in AWS

Closes #781 #779

### CODE REVIEW

File: pkg/kube/handler.go [hunk 1]
```diff
@@ -488,8 +489,12 @@ func (h *Handler) addNode(w http.ResponseWriter, r *http.Request) {
 	}
 
 	config := steps.NewConfig(k.Name, "", k.AccountName, kubeProfile)
+	config.ClusterName = k.Name
```
Comment (saheienko) [inline pkg/kube/handler.go]: Cluster name is passed to constructor and then it's set again.

Comment (stgleb) [reply pkg/kube/handler.go]: yeah, duplicate, thank you

File: pkg/model/kube.go [hunk 2]
```diff
@@ -35,6 +36,8 @@ type Kube struct {
 	HelmVersion            string     `json:"helmVersion"`
 	Networking             Networking `json:"networking"`
 
+	CloudSpecificSettings profile.CloudSpecificSettings `json:"cloudSpecificSettings" valid:"-"`
```
Comment (saheienko) [inline pkg/model/kube.go]: Could it be renamed to `CloudSpec`?

File: pkg/provisioner/provisioner.go [hunk 3]
```diff
@@ -83,17 +83,24 @@ func (tp *TaskProvisioner) ProvisionCluster(ctx context.Context,
 	err := tp.buildInitialCluster(ctx, profile, masters, nodes, config)
 
 	if err != nil {
-		return nil, errors.Wrap(err, "create initial cluster")
+		logrus.Errorf("create initial cluster %v", err)
```
Comment (saheienko) [inline pkg/provisioner/provisioner.go]: What kind of error it can return?

File: pkg/workflows/steps/amazon/delete_machine.go [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+func InitDeleteNode(fn GetEC2Fn) {
+	steps.RegisterStep(DeleteNodeStepName, &DeleteNodeStep{
+		GetEC2: fn,
+	})
+}
+
+func (s *DeleteNodeStep) Run(ctx context.Context, w io.Writer, cfg *steps.Config) error {
+	log := util.GetLogger(w)
+	logrus.Infof("[%s] - deleting node %s", s.Name(), cfg.Node.Name)
```
Comment (saheienko) [inline pkg/workflows/steps/amazon/delete_machine.go]: What logger is responsible for task logging?

File: pkg/workflows/steps/amazon/delete_machine.go [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+			InstanceIds: aws.StringSlice(instanceIDS),
+		})
+
+	if err != nil {
+		logrus.Error(ErrDeleteNode, err.Error())
+		return errors.Wrap(ErrDeleteNode, err.Error())
+	}
+
+	if err != nil {
+		logrus.Error(ErrDeleteNode, err.Error())
```
Comment (saheienko) [inline pkg/workflows/steps/amazon/delete_machine.go]: Would it be an informative message?

Comment (saheienko) [reply pkg/workflows/steps/amazon/delete_machine.go]: Remove this error check (duplication).

File: pkg/workflows/steps/amazon/delete_machine.go [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+	logrus.Debugf("Got %d instance outputs",
+		len(describeInstanceOutput.Reservations))
+	instanceIDS := make([]string, 0)
+	for _, res := range describeInstanceOutput.Reservations {
+		for _, instance := range res.Instances {
+			instanceIDS = append(instanceIDS, *instance.InstanceId)
+		}
+	}
+	if len(instanceIDS) == 0 {
+		logrus.Infof("[%s] - no nodes in k8s cluster %s", s.Name(), cfg.ClusterName)
```
Comment (saheienko) [inline pkg/workflows/steps/amazon/delete_machine.go]: Please, add an instance name to log message.
Are you searching for instance in k8s cluster?

File: pkg/workflows/steps/amazon/delete_machine.go (hunk 4 above)
Comment (egopher) [reply pkg/workflows/steps/amazon/delete_machine.go]: Should be task logger.

File: pkg/workflows/steps/amazon/delete_cluster.go [hunk 7]
```diff
@@ -40,7 +40,7 @@ func (s *DeleteClusterStep) Run(ctx context.Context, w io.Writer, cfg *steps.Con
 		Filters: []*ec2.Filter{
 			{
 				Name:   aws.String("tag:KubernetesCluster"),
```
Comment (egopher) [inline pkg/workflows/steps/amazon/delete_cluster.go]: Change it to clouds.ClusterIDTag

Comment (stgleb) [reply pkg/workflows/steps/amazon/delete_cluster.go]: Done

File: pkg/workflows/steps/amazon/delete_machine.go (hunk 4 above)
Comment (stgleb) [reply pkg/workflows/steps/amazon/delete_machine.go]: @egopher Done

**Pull request merged**